### ISSUE TITLE
nnUNetv2_plan_and_preprocess-- Issues for 2D datasets 

### ISSUE THREAD

**Issue by kabbas570** [MIC-DKFZ/nnUNet#1637]
I am using nnUNet for two different 2d datasets (MIT and MnM); the input has two channels, _0000 and _0001. It works fine for both datasets, but I want to know about the TWO issues. 

MIT has a spatial dimension of **1500 X 1500** MnM has a spatial dimension of **256  X 256.** 

I have converted them to raw data with these dimensions,
 **::::: For MIT Data   :::**
 10078675_15_0000.nrrd --> (1, 1500, 1500)
 10078675_15_0001.nrrd  --> (1, 1500, 1500)
 10078675_15.nrrd --> (1500, 1500)  ## for GT
 **::::: For MnM Data   :::**
 001_LA_ES_0000 --> (1, 256, 256) 
001_LA_ES_0001 --> (1, 256, 256) 
001_LA_ES--> (256, 256)   ## for GT

 However, nnUNet configures a 2d network with different shapes (order of axis)  for both 
**::::: For MIT Data   :::** 
Input shape-> (2,1,1500,1500)
 Label shape --> (1,1,1500,1500)
 **::::: For MnM Data   :::**
 Input shape --> (2,256,1,252)
 Label shape --> (1,256,1,252) 

1- Both can be trained without any issue, BUT the problem is the training with MnM Data is taking too long to train, like **three times** more times; 
Do you have any hint why this error arises? 
Or how to make the MnM dimensions the same as MIT data for training? 
2- The second issue I found for MnM data, the nnUNetv2_plan_and_preprocess even changes the label values and adds some extra **-1s** not sure from where they come. As shown in the below figure.

<img width="287" alt="image" src="https://github.com/MIC-DKFZ/nnUNet/assets/56618776/a545c3fb-2c93-440f-ba21-ccf3433d7d99">

 Thanks 
Cheers
 Abbas

**Comment by constantinulrich** [comment]
Hi, 
1. what is the file extension for your MnM dataset and could you send me both plan jsons?
2. -1 after preprocessing is ok. The will be removed during dataloading/augmentation

Best, Constantin

**Comment by kabbas570** [comment]
Dear Constantin,
Thanks for your reply.

I have solved the first issue, From here, 'nnU-Net transposes the input data so that the axes with the highest resolution (lowest spacing) come last. This is because the 2D U-Net operates on the trailing dimensions (more efficient slicing due to internal memory layout of arrays). Future work might move this setting to affect only individual configurations.'

I have resampled the data so that it has an identical voxel size and it shifted (2,256,1,252) to (2,1,256,252).

For the second, issue, I will l am getting the predictions which are fine, so I think the -1s are removed.

Thanks, Closing the issue.
